src/classiceditor.js

```diff
@@ -70,7 +70,12 @@ export default class ClassicEditor extends Editor {
 
 		this.model.document.createRoot();
 
-		this.ui = new ClassicEditorUI( this, new ClassicEditorUIView( this.locale, this.editing.view ) );
+		const shouldGroupWhenFull = this.config.get( 'toolbar.shouldGroupWhenFull' );
+		const view = new ClassicEditorUIView( this.locale, this.editing.view, {
+			shouldToolbarGroupWhenFull: shouldGroupWhenFull === undefined ? true : shouldGroupWhenFull
```

Comment:
TBH I hate this bit. I should've used `this.config.defined( 'toolbar.shouldGroupWhenFull', true )` in this `constructor()` instead. But... 

The thing is that toolbar configuration (as in the `Config` instance) can be either as 
* `toolbar: [ 'bold', ... ]` 
* or `toolbar: { items: [ 'bold', ... ], otherOptions: ... }` 
at this stage. 

Doing `this.config.define( 'toolbar.shouldGroupWhenFull', true )` overrides the former format (the later is OK) creating `toolbar: {  shouldGroupWhenFull: true }` and nuking the items configuration passed to the `create()`.

So to keep things working I'd have to do something like this:

```js
this.config.set( 'toolbar', normalizeToolbarConfig( this.config.get( 'toolbar' ) ) );
this.config.define( 'toolbar.shouldGroupWhenFull', true )
```

which I hate even more because it changes the internal form (as in the `Config` instance, not the plain object provided by the user) of the toolbar configuration and who knows with what consequences.

So yeah... maybe I'm overreacting but I think we should discuss this bit.

Reply:
I agree... that sucks. You can report a followup and comment it here with a link.

Also, I'd move the condition outside of the `ClassicEditorUIView()` constructor call to have the correct value directly in `shouldGroupWhenFull` variable.

Reply:
Also, see https://github.com/ckeditor/ckeditor5-core/pull/201/files#r347772936.